Source/MQTTnet/Implementations/MqttWebSocketChannel.cs

```diff
@@ -221,11 +221,20 @@ void SetupClientWebSocket(ClientWebSocket clientWebSocket)
 
 #if !NETSTANDARD1_3
 #if !WINDOWS_UWP
-            clientWebSocket.Options.UseDefaultCredentials = _options.UseDefaultCredentials;
+            if (_options.UseDefaultCredentials)
```

Comment:
I also crated a branch with a fix then found your PR. Basically it is the same but I added the following comment above line 224.

            // Only set the value if it is actually true. This property is not supported on all platforms
            // and will throw a _PlatformNotSupported_ (i.e. WASM) exception when being used regardless of the actual value.

Please add such a comment so that other know why we have to do it this way.

Then I will approve and merge the PR.